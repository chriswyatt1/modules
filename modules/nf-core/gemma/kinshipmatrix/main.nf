process GEMMA_KINSHIPMATRIX {
    tag "$meta.id"
    label 'process_single'

    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gemma:0.98.5--ha36d3ea_0':
        'biocontainers/gemma/0.98.1dfsg-1-deb' }"

    input:
    tuple val(meta), path(genotype)
    tuple val(meta2), path(phenotype)

    output:
    // TODO nf-core: Update the information obtained from bio.tools and make sure that it is correct
    tuple val(meta), path("*.{}"), emit: heat_map
    tuple val(meta), path("*.{}"), emit: quality_control_report
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    gemma \\
        $args \\
        -@ $task.cpus \\
        -g $genotype \\
        -p $phenotype  \\
        -gk
        -o $meta\.out

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gemma: \$(gemma --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch $meta\.out
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gemma: \$(gemma --version)
    END_VERSIONS
    """
}
