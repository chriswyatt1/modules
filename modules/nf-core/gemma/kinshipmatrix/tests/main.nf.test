nextflow_process {

    name "Test Process GEMMA_KINSHIPMATRIX"
    script "../main.nf"
    process "GEMMA_KINSHIPMATRIX"

    tag "modules"
    tag "modules_nfcore"
    tag "gemma"
    tag "gemma/kinshipmatrix"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("sarscov2 - bam") {

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('mouse_hs1940.geno.txt', checkIfExists: true),
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file('mouse_hs1940.pheno.txt', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('mouse_hs1940.geno.txt', checkIfExists: true),
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file('mouse_hs1940.pheno.txt', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
