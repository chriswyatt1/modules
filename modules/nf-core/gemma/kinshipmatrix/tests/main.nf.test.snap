{
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,02383d9d5046a267d81f99463df55749"
                ],
                "matrix": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,02383d9d5046a267d81f99463df55749"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-14T16:34:56.015094848"
    },
    "gwas - vcf - pheno - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out.cXX.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,02383d9d5046a267d81f99463df55749"
                ],
                "matrix": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out.cXX.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,02383d9d5046a267d81f99463df55749"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-14T17:07:00.03739859"
    },
    "sarscov2 - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.out.cXX.txt:md5,83fbef03ff6c4e7e89bebd52ce9d11a6",
                            "test.out.log.txt:md5,f7eccba3f5e053669b3fb2b8c5f635e7"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,02383d9d5046a267d81f99463df55749"
                ],
                "matrix": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.out.cXX.txt:md5,83fbef03ff6c4e7e89bebd52ce9d11a6",
                            "test.out.log.txt:md5,f7eccba3f5e053669b3fb2b8c5f635e7"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,02383d9d5046a267d81f99463df55749"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-14T16:34:38.506668318"
    },
    "gwas - vcf - pheno": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out.cXX.txt:md5,83fbef03ff6c4e7e89bebd52ce9d11a6"
                    ]
                ],
                "1": [
                    "versions.yml:md5,02383d9d5046a267d81f99463df55749"
                ],
                "matrix": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.out.cXX.txt:md5,83fbef03ff6c4e7e89bebd52ce9d11a6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,02383d9d5046a267d81f99463df55749"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-14T16:59:21.46538756"
    }
}